rule.go

```diff
@@ -51,8 +51,8 @@ func NewRule() *Rule {
 		SuppressIfgroup:   -1,
 		SuppressPrefixlen: -1,
 		Priority:          -1,
-		Mark:              -1,
-		Mask:              -1,
+		Mark:              0,
+		Mask:              0,
```

Comment:
Shouldn't the defaults be `0xFFFFFFFF` to preserve the existing behavior?

Reply:
[Complete test source code](https://gist.github.com/qxoqx/452b0174907fe616c650dbf1a4c76cb0)
```
AddRule(200, 20, -1, 0xFFFFFFFF)
AddRule(200, 21, 0, 0xFFFFFFFF)
AddRule(200, 22, 1, 0xFFFFFFFF)
AddRule(200, 23, 0xFFFFFFFF, 0xFFFFFFFF)
/*
	20:     from all fwmark 0 lookup 200
	21:     from all fwmark 0 lookup 200
	22:     from all fwmark 0x1 lookup 200
	23:     from all fwmark 0xffffffff lookup 200
*/
```
if mark == -1 will not set FRA_FWMARK, so the mask will be ignored, the phenomenon looks the same as mark == 0, if mask == 0xffffffff, the result is different.
		so the default value of mark should be 0, even I think rule.Mark >= 0 can be modified to rule.Mark > 0

```
AddRule(200, 24, 0xFFFFFFFF, 0)
AddRule(200, 25, 0xFFFFFFFF, 1)
AddRule(200, 26, 0xFFFFFFFF, -1)
AddRule(200, 27, 0xFFFFFFFF, 0xFFFFFFFF)
/*
	24:     from all fwmark 0xffffffff/0 lookup 200
	25:     from all fwmark 0xffffffff/0x1 lookup 200
	26:     from all fwmark 0xffffffff lookup 200
	27:     from all fwmark 0xffffffff lookup 200
*/
```
The above results look like, when the mask is -1, it is the same as when the mask is 0xFFFFFFFF,
		so the default value of the mask should be changed from -1 to 0xFFFFFFFF, but I found the following scenario
```
AddRule(200, 30, 1, -1)
AddRule(200, 31, 1, 0xFFFFFFFF)
AddRule(200, 32, -1, -1)
AddRule(200, 33, -1, 0xFFFFFFFF) // <----
AddRule(200, 34, 0, -1)
AddRule(200, 35, 0, 0xFFFFFFFF) // <----
/*
	30:     from all fwmark 0x1 lookup 200
	31:     from all fwmark 0x1 lookup 200
	32:     from all lookup 200
	33:     from all fwmark 0 lookup 200
	34:     from all lookup 200
	35:     from all fwmark 0 lookup 200
*/
```
When mark == 1, mask == -1 is the same as mask == 0xFFFFFFFF, but when mark == 0, mask == -1 is different from mask == 0xFFFFFFFF (pref 32,33), the same problem with mark == -1. This makes me very confused.

Reply:
I have resubmitted the code and made the following changes. Please review again.

rule.go
```
		Mark:              0,
		Mask:              0xFFFFFFFF,
```

rule_linux.go
``` 
	if rule.Mark > 0 {
		b := make([]byte, 4)
		native.PutUint32(b, uint32(rule.Mark))
		req.AddData(nl.NewRtAttr(nl.FRA_FWMARK, b))
	}
	if rule.Mask >= 0 && rule.Mask < 0xFFFFFFFF || (rule.Mask >= 0xFFFFFFFF && rule.Mark > 0) {
		b := make([]byte, 4)
		native.PutUint32(b, uint32(rule.Mask))
		req.AddData(nl.NewRtAttr(nl.FRA_FWMASK, b))
```